app/assets/javascripts/application.js

```diff
@@ -129,8 +137,22 @@ function mute() {
   });
 }
 
-function markAsRead(id) {
-  $.get( "/notifications/"+id+"/mark_as_read");
+function markReadSelected() {
+  if ( $(".js-table-notifications tr").length === 0 ) return;
+  marked = $(".js-table-notifications tr.active input:checked");
```

Comment:
This logic is now repeated several times in application.js.  I should DRY it up, but I haven't yet.